src/raft/system.ts

```diff
@@ -54,13 +53,12 @@ export function execute(command : string, args : string [], options? : ExecuteOp
         directoryCreated = Promise.resolve(); //No need to create directory
         raftlog(tag, "Running in the current working directory");
     }
-
     return directoryCreated.then(() => {
-        return Promise.fromNode((callback) => {
-            child_process.exec(cmdStr, nodeOptions, callback);
-        }, {multiArgs : true});
-    }).then((buffers : Buffer []) => {
-        raftlog(tag, "Finished successfullly");
-        return { stdout : buffers[0], stderr : buffers[1]};
+        return new Promise((resolve) => {
+            child_process.exec(cmdStr, nodeOptions as child_process.ExecOptionsWithBufferEncoding, (err : Error, stdout : Buffer, stderr : Buffer) => {
+                raftlog(tag, "Finished successfully");
+                resolve({stdout: stdout, stderr: stderr});
```

Comment:
You don't handle errors here. If there is an error you need to call the promises reject function.